shared/src/commonMain/kotlin/edu/usf/cutr/otp/plan/api/PlanApi.kt

```diff
@@ -0,0 +1,109 @@
+/*
+   Copyright University of South Florida 2021
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+package edu.usf.cutr.otp.plan.api
+
+import edu.usf.cutr.otp.ApplicationDispatcher
+import edu.usf.cutr.otp.plan.model.Planner
+import edu.usf.cutr.otp.plan.model.RequestParameters
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import io.ktor.client.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import kotlinx.serialization.json.Json
+
+class PlannerApi(private val host: String,
```

Comment:
For the constructors, can you please change them so they don't split the host and port, and instead take in a single String with the full path including the `/plan` part of the path?

So example input would be `https://10.0.2.2:8080/otp/routers/default/plan`, and we would just add the parameters when building the URL. 

This allows use of HTTPS (which is really important - we're sending user real-time locations as parameters), and we also can't assume that all OTP deployments will have the `/otp/routers` path. For example, see https://github.com/OneBusAway/onebusaway-android/issues/965. Sound Transit has an OTP path like http://tpng.api.soundtransit.org/tripplanner/st/plan. And they are migrating to `http://tpng.api.soundtransit.org/tripplanner/st/v1/plan`, which has a version indicator included.

Please also add Kotlin docs for the classes and public methods that explain what the expected input is.

Reply:
Thanks for letting me know! I'm not certain if the OTP allows `https` requests. I tried using on the browser and Android, the connection did not go through!

Reply:
HTTPS requires configuration with a cert. or a proxy so it probably won't work out-of-the-box - but we should still support it in the library.

Reply:
Should we create a self-signed certificate? Looks like `ktor` supports that. https://ktor.io/docs/self-signed-certificate.html

Reply:
We shouldn't need to worry about any of that in the client - it's only if you're running a server. So we just need to make sure in the client library that we support https URLs.

Reply:
Meaning, we need to allow the app to pass in a string that indicates http or https. And send the request using the appropriate protocol.

Reply:
I guess we allow the user to enter the `https` url (by not restricting them to HTTP) and it's up to what state of OTP they use if I'm not wrong?